### ISSUE TITLE
test.py seems to stall once reaching trainer.test(), printing "Testing: 0it [00:00, ?it/s]" to the console.

### ISSUE THREAD

**Issue by dongzhang04** [vitusbenson/greenearthnet#1]
Hi Vitus Benson,

I'm Dong Zhang, the student working under Richard Fernandes for the next 4 months. 

To begin with, I'm running mostly the same environment outlined in the installation section of the readme, however I upgraded pytorch-lightning to version 1.9.5 and subsequently torchmetrics to version 0.11.4 due to issues with installing earthnet-models-pytorch in which the installer required lightning >= 1.9. 

When running test.py with the contextformer6M/seed=97 configuration on the ood-t_chopped data, I had initially encountered an issue where the execution would stall at the instantiation of the trainer [when calling pl.Trainer()], however I believe this was due to an issue with my machine's gpu and/or the environment's cuda installation as including the code: os.environ["CUDA_VISIBLE_DEVICES"] = ""
was able to further the execution of the script when using cpus and not gpus. The following code was also added/modified in order to set the script to use only cpus:
    if "gpus" in setting_dict["Trainer"]:
        del setting_dict["Trainer"]["gpus"]
        setting_dict["Trainer"]["accelerator"] = "cpu"
        # setting_dict["Trainer"]["devices"] = 1

After implementing these changes [along with some print statements], running the script would result in the following output, which is where I am currently lost: 
<img width="1120" height="575" alt="Image" src="https://github.com/user-attachments/assets/d9b7c43e-8dcc-4393-aa0d-8277fad92a6c" />

For some added background, I've tried running a simpler script [taken from https://pytorch-lightning.readthedocs.io/en/1.5.10/common/trainer.html] to test if the issue persists elsewhere: 
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.utils.data import DataLoader, random_split
from torchvision.datasets import MNIST
import torchvision.transforms as transforms
import torch.distributed as dist
import os

class MNISTClassifier(nn.Module):
    def __init__(self):
        super().__init__()
        self.layer_1 = torch.nn.Linear(28 * 28, 128)
        self.layer_2 = torch.nn.Linear(128, 10)

    def forward(self, x):
        x = x.view(x.size(0), -1)
        x = self.layer_1(x)
        x = F.relu(x)
        x = self.layer_2(x)
        return x

if global_rank == 0:
    mnist_train = MNIST(os.getcwd(), train=True, download=True)
    mnist_test = MNIST(os.getcwd(), train=False, download=True)

dist.barrier()

transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Normalize((0.1307,), (0.3081,))
])
mnist_train = MNIST(os.getcwd(), train=True, transform=transform)
mnist_test = MNIST(os.getcwd(), train=False, transform=transform)

mnist_train, mnist_val = random_split(mnist_train, [55000, 5000])
mnist_test = MNIST(os.getcwd(), train=False, download=True)

mnist_train = DataLoader(mnist_train, batch_size=64)
mnist_val = DataLoader(mnist_val, batch_size=64)
mnist_test = DataLoader(mnist_test, batch_size=64)
pytorch_model = MNISTClassifier()
optimizer = torch.optim.Adam(pytorch_model.parameters(), lr=1e-3)

def cross_entropy_loss(logits, labels):
    return F.nll_loss(logits, labels)

num_epochs = 1
for epoch in range(num_epochs):
    for train_batch in mnist_train:
        x, y = train_batch
        logits = pytorch_model(x)
        loss = cross_entropy_loss(logits, y)
        print('train loss: ', loss.item())
        loss.backward()
        optimizer.step()
        optimizer.zero_grad()

with torch.no_grad():
    val_loss = []
    for val_batch in mnist_val:
        x, y = val_batch
        logits = pytorch_model(x)
        val_loss.append(cross_entropy_loss(logits, y).item())

and in this case, the script stalls after reaching the loss.backward() call of the first train_batch. I believe that a similar issue may be occurring when running test.py, seeing as the script also stalls on the 0th iteration of the training [based off the output seen in the earlier image] however I'm struggling to find a cause. 

Apologies for such late notice on the issue.

**Comment by vitusbenson** [comment]
Hi @dongzhang04 

If I understood your problem correctly, the struggle seems to be that the old version of `lighting` & `torchmetrics` create problems for you.

To address this, I've pushed a few fixes, such that the code can now run with the latest versions. I've also simplified the install instructions.

I recommend that you:
1. Create a fresh python environment following the new instructions
2. Pull the latest changes to `greenearthnet`
3. Then run `test.py` as follows: 
```bash
CUDA_VISIBLE_DEVICES="" python test.py model_configs/contextformer/contextformer6M/seed=42.yaml model_weights/contextformer/contextformer6M/seed=42.ckpt --track ood-t_chopped --pred_dir model_preds/contextformer/contextformer6M/seed=42/ --data_dir /path/to/earthnet2021x/ 
```

Cheers
Vitus

**Comment by dongzhang04** [comment]
Thanks! I was able to get it working with these changes!
